test/spmd/tensor/test_tensor_ops.py

```diff
@@ -95,6 +95,29 @@ def test_ones_like(self):
         ones_expected = torch.ones(4, 8)
         self.assertEqual(ones_expected, ones_like_dt.to_local())
 
+    @with_comms
+    def test_softmax(self):
+        device_mesh = DeviceMesh(
+            self.device_type, list(range(self.world_size))
+        )
+        input = torch.rand(8, 12, 16, device=self.device_type)
+        shard0_spec = Shard(0)
+        shard1_spec = Shard(1)
```

Comment:
Curious why we need to test this if we already have it in the list for the dtensor_ops_db test?
nit: shard1_spec and shard2_spec are unused..

Reply:
@anj-s sorry I'm not familiar with the dtensor_ops_db test but one example I think appropriate to refer is https://github.com/pytorch/tau/blob/6bbe0872aeb8faa8fab73862bae9d3f806ec7836/test/spmd/tensor/test_dtensor_ops.py#L180

`bmm` is in the list however it's still tested in https://github.com/pytorch/tau/blob/6bbe0872aeb8faa8fab73862bae9d3f806ec7836/test/spmd/tensor/test_dtensor_ops.py#L180

Does this resolve your concern? 

> nit: shard1_spec and shard2_spec are unused..
Thx for catching it! I plan to add tests for the whole space of (batch dim, softmax dim) combination. e.g. [0, 1, 2, -1] $\times$ [0, 1, 2]

Reply:
agreed, @XilunWu could you enable this by deleting `xfail("softmax")` in the `test_densor_ops.py`? it will test forward automatically for you, you can probably change this test to only test backward.

Reply:
QQ, so since we have not supported softmax for sharding dim. Will this be a problem for `test_dtensor_ops.py`?

Reply:
I tried running `pytest test/spmd/tensor/test_dtensor_ops.py -s -k softmax` and the "softmax" test is currently passing. The set of parameters tested is as follows:
| Tensor Dim  | Softmax Dim | Sharding |
| ------------- | ------------- | ------------- | 
| 0D (scalar) | 0 | Replicate |
| 1D  | 0  | Replicate |
| 2D  | {0, -1}  | Replicate |
| 3D | {2} | Replicate |

And the current "test_softmax" parameter set is:
| Tensor Dim  | Softmax Dim | Sharding |
| ------------- | ------------- | ------------- | 
| 3D | {0, 1, 2, -1} | {0, 1, 2, -1} |

My question is, since `test_softmax` in `test_tensor_ops.py` is not a duplicate to the "softmax" test in `test_dtensor_ops.py`, should we keep it?